### PR TITLE
Make nomad script GCP compatible again

### PR DESCRIPTION
**Pull request by nsmith5**
:gear: **Issue**

Some GCP related details got removed in the refactor from the shared nomad script.

:white_check_mark: **Fix**

While this doesn't add support for Nomad clients in GCP, it makes the shared script GCP compatible once again by adding back the components that used to be there so that it will be easier to add GCP in the future.

:question: **Tests**

- [x] Passed shellcheck
- [x] Nomad AWS still works

### CODE REVIEW

File: shared/nomad-scripts/nomad-startup.sh.tpl [hunk 1]
```diff
@@ -58,14 +66,16 @@ sleep 5
...
 echo "--------------------------------------"
-export aws_instance_metadata_url="http://169.254.169.254"
-export PUBLIC_IP="$(curl $aws_instance_metadata_url/latest/meta-data/public-ipv4)"
-export PRIVATE_IP="$(curl $aws_instance_metadata_url/latest/meta-data/local-ipv4)"
-if ! (echo $PUBLIC_IP | grep -qP "^[\d.]+$"); then
-    echo "Setting the IPv4 address below in /etc/circleci/public-ipv4."
-    echo "This address will be used in builds with \"Rebuild with SSH\"."
-    mkdir -p /etc/circleci
-    echo $PRIVATE_IP | tee /etc/circleci/public-ipv4
+if [ $CLOUD_PROVIDER == "AWS" ]; then
```
Comment (robinsturmeit) [inline shared/nomad-scripts/nomad-startup.sh.tpl:69]: I know that you are just restoring the old state here, but we will need to fix this for GCP, too, at some stage

**Review by robinsturmeit — APPROVED** [review]

**Review by lloydChris — COMMENTED** [review]
was this tested by launching nomad clients in GCP?

File: shared/nomad-scripts/nomad-startup.sh.tpl (hunk 1 above)
Comment (nsmith5) [reply shared/nomad-scripts/nomad-startup.sh.tpl:69]: Yeah agreed, but lets do that when we play the GCP support ticket perhaps?

**Comment by nsmith5** [comment]
Going to just make this AWS specific and remove from shared directory

**Pull request closed without merging**